trlx/utils/modeling.py

```diff
@@ -1,11 +1,129 @@
-from typing import MutableMapping, Union
+from typing import MutableMapping, Tuple, Union
 
+import functools
 import torch
+import torch.nn as nn
 import torch.nn.functional as F
 import torch.distributed as dist
+import transformers
 from typing import Tuple
 
 
+def make_head(n_embd: int, out: int) -> nn.Sequential:
+    """Returns a generic sequential linear head."""
+    return nn.Sequential(
+        nn.Linear(n_embd, n_embd * 2),
+        nn.ReLU(),
+        nn.Linear(n_embd * 2, out),
+    )
+
+
+# HuggingFace utilities
+
+
+def rhasattr(obj, attr):
+    """A chain-able attribute version of hasattr. For example, to check if
+    `obj` has the attribute `foo.bar.baz`, you can use:
+        `rhasattr(obj, "foo.bar.baz")`
+    Reference: https://stackoverflow.com/a/31174427
+    """
+    _nested_attrs = attr.split(".")
+    _curr_obj = obj
+    for _a in _nested_attrs[:-1]:
```

Comment:
Why [:-1]?

Reply:
For some weird reason, nesting `hasttr`s doesn't always identify the attributes correctly. This walks through the dot-accesses to find the next to last object attribute that can be used to grab the proper boolean `hasattr` value.